### PR TITLE
improve openapi import

### PR DESCRIPTION
**Pull request by calvernaz**
This PR improves OpenAPI import by adding support for `responseParameters`, `responseModels`, `models` resources and other minor fixes.

### CODE REVIEW

File: localstack/services/apigateway/helpers.py [hunk 1]
```diff
@@ -108,6 +106,78 @@ def __init__(self):
...
+        self.document = document
+        self.allow_recursive = allow_recursive
+        # cache which maps known refs to part of the document
+        self._cache = {}
+        self._refpaths = ["#"]
+
+    def _is_ref(self, item):
+        return isinstance(item, dict) and "$ref" in item
+
+    def _is_internal_ref(self, refpath):
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:120]: Would be helpful to add some type hints to the method parameters / return types in this class. 👍

File: localstack/services/apigateway/helpers.py [hunk 2]
```diff
@@ -108,6 +106,78 @@ def __init__(self):
...
+            return self._cache[refpath]
+
+        with self._pathctx(refpath):
+            if self._is_internal_ref(self.current_path):
+                cur = self.document
+            else:
+                raise NotImplementedError("External references not yet supported.")
+
+            for step in self.current_path.split("/")[1:]:
+                cur = cur[step]
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:150]: nit: We could use `cur.get(step)` here to hedge against `KeyError`, and then introduce a custom exception (e.g., `UnresolvableRefException`) that callers can catch explicitly. 

The other option would be to handle these cases gracefully and return `None` from the method if a ref cannot be resolved. 

I'd chose one of the two (instead of bubbling up raw `KeyError`s), to make the interface a bit more clear/explicit.

File: localstack/services/apigateway/helpers.py [hunk 3]
```diff
@@ -1186,42 +1261,65 @@ def add_path(path, parts, parent_id=""):
...
+                ),
                 content_handling=None,
             )
-            child.resource_methods[m]["methodIntegration"] = integration
+            child.resource_methods[method]["methodIntegration"] = integration
 
         rest_api.resources[child_id] = child
         return child
 
+    if definitions := resolved_schema.get("definitions"):
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:1306]:
```suggestion
    if definitions := resolved_schema.get("definitions", {}):
```

Comment (whummer) [reply localstack/services/apigateway/helpers.py:1306]: Better hedge against the case where `"definitions"` is potentially not present.

File: tests/unit/utils/test_common.py [hunk 4]
```diff
@@ -281,3 +281,9 @@ def forward_request(self, method, path, data, headers):
     # clean up
     proxy.stop()
     rm_rf(tmp_file)
+
+
+def load_test_resource(file_name: str, file_path: str = None) -> str:
```
Comment (whummer) [inline tests/unit/utils/test_common.py:286]: nit: I think this file name is a bit misleading - `test_common.py` is currently used to test the common utils (i.e., not to define utility functions that can be imported from elsewhere). I'd probably move this function into a file `tests/helpers.py`, or into our existing util file `localstack/utils/testutil.py` that already contains some test utilities. 👍

File: tests/unit/test_apigateway.py [hunk 5]
```diff
@@ -366,3 +372,55 @@ def test_render_custom_template(self, template):
...
             "stageVariable2": "value2",
         }
+
+
+def test_import_swagger_api():
+    api_spec = load_test_resource("openapi.swagger.json")
+    api_spec_dict = json.loads(api_spec)
+
+    # patch integration model
+    apigateway_starter.apply_patches()
```
Comment (whummer) [inline tests/unit/test_apigateway.py:382]: Not sure we want to establish this as a pattern - most of our patches are currently written in a non-idempotent way and calling them directly from a unit test may have undesired side effects. We usually write unit tests only for generic logic that does not depend on any of the actual service providers.

Let's better move this test to `tests/integration/...` where we can safely assume that the patches have already been properly applied.. 👍

File: localstack/services/apigateway/helpers.py [hunk 6]
```diff
@@ -108,6 +106,78 @@ def __init__(self):
         self.client_certificates = {}
 
 
+class Resolver:
```
Comment (whummer) [inline localstack/services/apigateway/helpers.py:109]: This is a pretty neat utility class! 👍 Should we add a couple of small unit tests, to cover the most basic functionality? (and maybe also some edge cases, like unresolvable references, ...)

**Review by whummer — COMMENTED** [review]
Nice changes @calvernaz , 👍  looking forward to seeing the schema resolver in action! Added a few smaller comments/suggestions..

File: localstack/services/apigateway/helpers.py (hunk 3 above)
Comment (calvernaz) [reply localstack/services/apigateway/helpers.py:1306]: We established an agreement, that is not required.

Comment (whummer) [reply localstack/services/apigateway/helpers.py:1306]: Yep, this comment can be ignored. Totally missed the guard on line 1306 - thanks for pointing it out. 👍

File: localstack/services/apigateway/helpers.py (hunk 2 above)
Comment (calvernaz) [reply localstack/services/apigateway/helpers.py:150]: Good points for sure, I made the modifications to hedge against the lookup and that is sufficient because we want the unresolvable references to be `None`. If you notice in this method the lookup is only supported for local references (no external references) and if the `current_path` (e.g `#/definitions/NotFound`) is not in the document, with the new lookup, it will return `None` and we proceed with the rest of the document.

File: tests/unit/utils/test_common.py (hunk 4 above)
Comment (calvernaz) [reply tests/unit/utils/test_common.py:286]: I'll keep it in the test suite then, because of the relative paths and until it because an issue maintaining, having it in the class seems to be the pattern to load these files anyway.

**Review by whummer — APPROVED** [review]
🚀
